service/controller/chart/v1/resource/releasemigration/create.go

```diff
@@ -0,0 +1,71 @@
+package releasemigration
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/giantswarm/microerror"
+	"github.com/giantswarm/operatorkit/controller/context/reconciliationcanceledcontext"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+	"github.com/giantswarm/chart-operator/service/controller/chart/v1/key"
+)
+
+// EnsureCreated ensures resources are migrated to helm v3.
+func (r *Resource) EnsureCreated(ctx context.Context, obj interface{}) error {
+	cr, err := key.ToCustomResource(obj)
+	if err != nil {
+		return microerror.Mask(err)
+	}
+
+	hasConfigMap, err := r.findHelmV2ConfigMaps(ctx, key.ReleaseName(cr))
+	if err != nil {
+		return microerror.Mask(err)
+	}
+
+	hasSecret, err := r.findHelmV3Secrets(ctx, key.ReleaseName(cr), key.Namespace(cr))
+	if err != nil {
+		return microerror.Mask(err)
+	}
+
+	// If Helm v2 release configmap had not been deleted and Helm v3 release secret is there,
+	// It means helm release migration is in progress.
+	if hasConfigMap && hasSecret {
+		r.logger.LogCtx(ctx, "level", "debug", "message", fmt.Sprintf("release %#q is under helmV3 migration", key.ReleaseName(cr)))
+		r.logger.LogCtx(ctx, "level", "debug", "message", "canceling reconciliation")
+		reconciliationcanceledcontext.SetCanceled(ctx)
+		return nil
+	}
+
+	r.logger.LogCtx(ctx, "level", "debug", "message", fmt.Sprintf("no pending migration on release %#q", key.ReleaseName(cr)))
+
+	return nil
+}
+
+func (r *Resource) findHelmV2ConfigMaps(ctx context.Context, releaseName string) (bool, error) {
```

Comment:
These functions are the same as `tillermigration` resources but not sure where I can keep them for general use.